tests/unit/ApiKeyRequestMiddlewareTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Sminnee\ApiKey\UnitTests;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Security\Member;
+use Sminnee\ApiKey\MemberApiKey;
+use Sminnee\ApiKey\ApiKeyMemberExtension;
+use Sminnee\ApiKey\ApiKeyRequestMiddleware;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Security\Security;
+
+class ApiKeyRequestMiddlewareTest extends SapphireTest
+{
+    /**
+     * @var Member
+     */
+    private $member;
+
+    /**
+     * @var MemberApiKey
+     */
+    private $key;
+
+    /**
+     * @var string
+     */
+    private $headerName;
+
+    /**
+     * @var string
+     */
+    protected static $fixture_file = 'ApiKeyRequestMiddlewareTest.yml';
+
+    /**
+    * @var array
+    */
+    protected static $required_extensions = [
+        Member::class => [
+            ApiKeyMemberExtension::class,
+        ],
+    ];
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->member = $this->objFromFixture(Member::class, 'admin');
+        $this->key = MemberApiKey::createKey($this->member->ID);
+        $this->headerName = Config::inst()->get(ApiKeyRequestMiddlware::class, 'header_name');
+    }
+
+    public function testPassthroughProcess()
+    {
+        // pass through with no login
+        $request = new HTTPRequest('GET', Director::absoluteBaseURL() . '/api/v1');
+        $middleware = new ApiKeyRequestMiddleware();
+
+        $calls = 0;
+        $delegate = function () use (&$calls) {
+            $calls++;
+        };
+        $middleware->process($request, $delegate);
```

Comment:
I wonder whether passing a mock in with something like `$this->expects($this->once())->method(...` might be cleaner - e.g. https://stackoverflow.com/a/33828931/2812842

Thoughts? This would apply to the other instances in this class too

Reply:
this is a neat trick - cheers 👍